File: LeetcodeProblems/blind-75/graph/longest-consecutive-sequence/solution.js
// Longest Consecutive Sequence - LeetCode #128

/**
 * @param {number[]} nums
 * @return {number}
 */
var longestConsecutive = function(nums) {
    const set = new Set(nums);
    let maxLength = 0;
    
    for (let num of set) {
        if (!set.has(num - 1)) {
            let current = num;
            let length = 1;
            
            while (set.has(current + 1)) {
                current++;
                length++;
            }
            
            maxLength = Math.max(maxLength, length);
        }
    }
    
    return maxLength;
};

File: LeetcodeProblems/blind-75/graph/longest-consecutive-sequence/solution.py
# Longest Consecutive Sequence - LeetCode #128

from typing import List

class Solution:
    def longestConsecutive(self, nums: List[int]) -> int:
        num_set = set(nums)
        max_length = 0
        
        for num in num_set:
            if num - 1 not in num_set:
                current = num
                length = 1
                
                while current + 1 in num_set:
                    current += 1
                    length += 1
                
                max_length = max(max_length, length)
        
        return max_length
